aten/src/ATen/native/vulkan/Vulkan.h

```diff
@@ -451,6 +463,40 @@ class ComputeUnit final {
   VkShaderModule computeShaderModule_;
 };
 
+class ComputeUnitFactory {
+ public:
+  ComputeUnitFactory(const VkDevice& device);
```

Comment:
It's generally considered good practice in C++ to mark single argument constructors as explicit to avoid implicit type conversions unless such behavior (i.e. in this case implicit conversion of VkDevice to ComputeUnitFactory) is desired.